istioctl/pkg/multicluster/remote_secret.go

```diff
@@ -255,36 +291,38 @@ func getOrCreateServiceAccountSecret(
 
 	// manually specified secret, make sure it references the ServiceAccount
 	if opt.SecretName != "" {
-		secret, err := client.CoreV1().Secrets(opt.Namespace).Get(ctx, opt.SecretName, metav1.GetOptions{})
+		secret, err := client.Kube().CoreV1().Secrets(opt.Namespace).Get(ctx, opt.SecretName, metav1.GetOptions{})
 		if err != nil {
 			return nil, fmt.Errorf("could not get specified secret %s/%s: %v",
 				opt.Namespace, opt.SecretName, err)
 		}
-		if err := validateServiceAccountSecret(serviceAccount, secret); err != nil {
+		if err := secretReferencesServiceAccount(serviceAccount, secret); err != nil {
 			return nil, err
 		}
 		return secret, nil
 	}
 
 	// first try to find an existing secret that references the SA
 	// TODO will the SA have any reference to secrets anymore, can we avoid this list?
-	allSecrets, err := client.CoreV1().Secrets(opt.Namespace).List(ctx, metav1.ListOptions{})
+	allSecrets, err := client.Kube().CoreV1().Secrets(opt.Namespace).List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("failed listing secrets in %s: %v", opt.Namespace, err)
 	}
 	for _, item := range allSecrets.Items {
 		secret := item
-		if validateServiceAccountSecret(serviceAccount, &secret) == nil {
+		if secretReferencesServiceAccount(serviceAccount, &secret) == nil {
 			return &secret, nil
 		}
 	}
 
 	// finally, create the sa token secret manually
 	// https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-service-account-api-token
 	// TODO ephemeral time-based tokens are preferred; we should re-think this
-	return client.CoreV1().Secrets(opt.Namespace).Create(ctx, &v1.Secret{
+	log.Infof("Creating token secret for service account %q", serviceAccount.Name)
+	secretName := tokenSecretName(serviceAccount.Name)
+	return client.Kube().CoreV1().Secrets(opt.Namespace).Create(ctx, &v1.Secret{
```

Comment:
@stevenctl does this `client.CoreV1().Secrets` -> `client.Kube().CoreV1().Secrets` work as expected without the underlying change to `pkg/kube/client.go` from the original PR?
https://github.com/istio/istio/pull/38789/files#diff-137bba79988de442c1fde0fc29520d6d7e5a5239ceac82f5439a5bf513ebb2d4

Reply:
Nope, I think https://github.com/istio/istio/pull/38789/files#diff-137bba79988de442c1fde0fc29520d6d7e5a5239ceac82f5439a5bf513ebb2d4L111

Line 111 in the "before" part of the diff already had the `Kube()` method returning the embedded client.

Reply:
So no behavior change.